code/modules/mob/animations.dm

```diff
@@ -156,18 +156,26 @@ note dizziness decrements automatically in the mob's Life() proc.
 	animate(src, pixel_x = pixel_x + pixel_x_diff, pixel_y = pixel_y + pixel_y_diff, time = 2)
 	animate(pixel_x = default_pixel_x, pixel_y = default_pixel_y, time = 2)
 
-/mob/do_attack_animation(atom/A)
+/mob/do_attack_animation(atom/A, atom/movable/weapon)
 	..()
 	is_floating = 0 // If we were without gravity, the bouncing animation got stopped, so we make sure we restart the bouncing after the next movement.
 
-	// What icon do we use for the attack?
-	var/image/I
-	if(hand && l_hand) // Attacked with item in left hand.
-		I = image(l_hand.icon, A, l_hand.icon_state, A.layer + 1)
-	else if (!hand && r_hand) // Attacked with item in right hand.
-		I = image(r_hand.icon, A, r_hand.icon_state, A.layer + 1)
-	else // Attacked with a fist?
-		return
+	// What are we attacking with?
+	if(!weapon)
+		weapon = get_active_hand()
+		if(!weapon)
+			return
+
+	// Create an image to show to viewers.
+	// Reset plane and layer so that it doesn't inherit the UI settings from equipped items.
+	var/image/I = new(loc = A)
```

Comment:
What code were you using w/ MAs when you tried them?

Reply:
I don't recall exactly, I think it was more or less the same but `var/mutable_appearance/MA = new(A)`.

Reply:
Should be:
```dm
var/mutable_appearance/MA = new(weapon)
MA.loc = A
```
(with the `appearance =` line deleted)